src/Paprika/Merkle/Hashing.cs

```diff
@@ -0,0 +1,68 @@
+using Paprika.Crypto;
+using Paprika.Data;
+using Paprika.RLP;
+
+namespace Paprika.Merkle;
+
+public static partial class Node
+{
+    public ref partial struct Leaf
+    {
+        public static KeccakOrRlp KeccakOrRlp(NibblePath nibblePath, Account account) =>
+            KeccakOrRlp(nibblePath, account, Keccak.OfAnEmptyString, Keccak.EmptyTreeHash);
+
+        public static KeccakOrRlp KeccakOrRlp(
+            NibblePath nibblePath,
+            Account account,
+            Keccak codeHash,
+            Keccak storageRootHash
+        )
+        {
+            // Stage 1: value = RLP(accountBalance, accountNonce, codeHash, storageRootHash)
+            var rlpStream = RlpOfAccount(account, codeHash, storageRootHash);
+
+            // Stage 2: res = KeccakOrRlp(nibblePath, value)
+            var result = Encode(nibblePath, rlpStream.Data);
+
+            return result;
+        }
+
+        private static RlpStream RlpOfAccount(Account account, Keccak codeHash, Keccak storageRootHash)
+        {
+            var contentLength =
+                Rlp.LengthOf(account.Balance)
+                + Rlp.LengthOf(account.Nonce)
+                + Rlp.LengthOfKeccakRlp // CodeHash
+                + Rlp.LengthOfKeccakRlp; // StorageRootHash
+
+            Span<byte> rlpBuffer = new byte[Rlp.LengthOfSequence(contentLength)];
```

Comment:
Let's avoid allocations using `stackalloc`.

Reply:
`stackalloc byte[Rlp.LengthOfSequence(contentLength)]` it is, isn't it? 👁️

Reply:
We cannot do that since the `rlpBuffer` is wrapped into a `RlpStream` which we return at the end of the function.